lib/pry/input_completer.rb

```diff
@@ -239,4 +227,30 @@ def build_path(input)
     end
     return path, input
   end
+  
+    private
+
+    def self.fetch_obj_space_collection
+      (self.obj_spase_mutex ||= Mutex.new).synchronize do
```

Comment:
As variant I may do global Mutex instance, but it really does not, matter a lot in this chunk. Main Idea is to avoid double usage of ObjectSpace.